util/array.c

```diff
@@ -19,3 +19,34 @@ size_t push_zeroes_to_end(uint32_t arr[], size_t n) {
 
 	return ret;
 }
+
+// Values MUST be greater than 0
+bool set_add(uint32_t values[], uint32_t* len, uint32_t cap, uint32_t target) {
+	if (*len == cap) {
+		return false;
+	}
+	assert(target > 0);
+	for (uint32_t i = 0, n = *len; i < n; ++i) {
+		if (values[i] == target) {
+			return false;
+		}
+	}
+	values[(*len)++] = target;
+	return false;
+}
+
+// Values MUST be greater than 0
+bool set_remove(uint32_t values[], uint32_t* len, uint32_t cap, uint32_t target) {
+	if (*len == 0) {
+		return false;
+	}
+	for (uint32_t i = 0, n = *len; i < n; ++i) {
+		if (values[i] == target) {
+			uint8_t last_elem_pos = --(*len);
```

Comment:
This should be `size_t`

Reply:
oh whoops. remnant from the small set version.